manifests/virtual.pp

```diff
@@ -1,6 +1,17 @@
 class amanda::virtual {
   include amanda::params
 
+  if $amanda::client::ensure == 'absent' {
+    $client_ensure = 'absent'
+  } else {
+    $client_ensure = 'present'
+  }
+  if $amanda::server::ensure == 'absent' {
+    $server_ensure = 'absent'
+  } else {
+    $server_ensure = 'present'
+  }
```

Comment:
In order to access `$amanda::client::ensure` and `$amanda::server::ensure` we must `include amanda::client` and `include amanda::server` first.

Reply:
Well, I'm note sure about this. I've tested (with Puppet 7), and the additional includes doesn't seem to be needed. Also, adding these includes changes the Amanda module behavior as the defaults are set to 'present'.

Reply:
Well my point is that if the class `amanda::server` is not included, `$amanda::server::ensure` will be `undef` while the default value is 'present' when `amanda::server` is included… that's weird and seems fragile:

```puppet
class { 'amanda::client':
}
# amanda::client include amanda
# `-> amanda include amanda::virtual
#     `-> amanda::virtual sees $amanda::server::ensure == undef
#                          and $amanda::client::ensure == 'present'
class { 'amanda::server':
}
```

```puppet
class { 'amanda::server':
}
# amanda::server include amanda
# `-> amanda include amanda::virtual
#     `-> amanda::virtual sees $amanda::server::ensure == 'present'
#                          and $amanda::client::ensure == undef
class { 'amanda::client':
}
```

```puppet
class { 'amanda':
}
# amanda include amanda::virtual
# `-> amanda::virtual sees $amanda::server::ensure == undef
#                      and $amanda::client::ensure == undef
class { 'amanda::client':
}
class { 'amanda::server':
}
```

I have no experience with amanda, but this does not look right :fearful: